add deserialization and transform request image file

YiminArava4508 opened the pull request:
the API should now deserialize the model object and transform the user's image file to a NumPy array for the ml model to predict a response

YiminArava4508 committed:
Update main.py

YiminArava4508 committed:
Merge branch 'main' of https://github.com/YiminArava4508/image-classifier-example into main

YiminArava4508 committed:
Merge branch 'main' of https://github.com/YiminArava4508/image-classifier-example into main

YiminArava4508 committed:
Deserialize model object

the API should now deserialize the model object and transform the user's image into a numpy array so the model can predict a response

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
+
+model_test = model_from_json(json_model)
+
+# Some functions for resp ---------------------------------------------------------------------------------------------
```
@YiminArava4508 There's a thing called "docstrings" when you put your comments and notes for a method into the method itself, check out this doc: https://www.python.org/dev/peps/pep-0257/

So for this one, it would look like this:

```python
def result(array):
    """
    Given the prediction array from the model, this method returns the predicted label.
    """
    if array[0][0] > array[0][1]:
        return 'genuine'
    elif array[0][0] < array[0][1]:
        return 'forged'
    else:
        return 'unable to determine'
```

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
+        return 'forged'
+    else:
+        return 'unable to determine'
+def result_percentage(array):
```
Add a docstring here too and make sure there's a blank line in between the methods

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
-
+# Imports ------------------------------------------------------------------------------------------------------------
```
We can remove this

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
+# Imports ------------------------------------------------------------------------------------------------------------
 from app import app
 from flask import Flask, request, redirect, jsonify, send_from_directory
+import tensorflow as tf
```
Remove this, the import isn't used

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
-
+# Imports ------------------------------------------------------------------------------------------------------------
 from app import app
 from flask import Flask, request, redirect, jsonify, send_from_directory
```
You can remove the "redirect" import, it's not used anywhere

JasonFreeberg commented on backend/image-upload-api/flask-api/main.py:
```diff
@@ -1,42 +1,78 @@
 from app import app
 from flask import Flask, request, redirect, jsonify, send_from_directory
+import tensorflow as tf
+from tensorflow import keras
+import numpy as np
+import json
+from PIL import Image,  ImageOps
```
Tensorflow, keras, numpy, and PIL need to be added to the `requirements.txt` file. You can update the requirements.txt file using `pip freeze > requirements.txt`

YiminArava4508 committed:
Clean up useless code and update requirement.txt

removed unused imports, added doc strings, removed redundant comments, and updated requirements.txt

YiminArava4508 committed:
Updated main.py

removed several trailing whitespaces and added module docstrings to functions

YiminArava4508 committed:
Update main.py

removed unnecessary "else" after "return"

YiminArava4508 commented:
I removed some unnecessary code based on the pylint CI, the code is much cleaner now. The rest of the checks are quite weird I am not sure if I should follow the suggestions
![Capture](https://user-images.githubusercontent.com/85930454/132923089-0d199089-3dfd-4a53-8e7c-b7043f7cb9d6.PNG)

JasonFreeberg commented:
Yeah don't worry about those last ones, there's something wrong with the CI or how the project files are organized. I'll fix that.

I'll review it tomorrow and merge it in

Commit:
Update main.py
